server/src/main/java/org/opensearch/search/backpressure/stats/SearchShardTaskStats.java

```diff
@@ -30,21 +31,29 @@
 public class SearchShardTaskStats implements ToXContentObject, Writeable {
     private final long cancellationCount;
     private final long limitReachedCount;
+    private final long completionCount;
     private final Map<TaskResourceUsageTrackerType, TaskResourceUsageTracker.Stats> resourceUsageTrackerStats;
 
     public SearchShardTaskStats(
         long cancellationCount,
         long limitReachedCount,
+        long completionCount,
         Map<TaskResourceUsageTrackerType, TaskResourceUsageTracker.Stats> resourceUsageTrackerStats
     ) {
         this.cancellationCount = cancellationCount;
         this.limitReachedCount = limitReachedCount;
+        this.completionCount = completionCount;
         this.resourceUsageTrackerStats = resourceUsageTrackerStats;
     }
 
     public SearchShardTaskStats(StreamInput in) throws IOException {
         this.cancellationCount = in.readVLong();
         this.limitReachedCount = in.readVLong();
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            this.completionCount = in.readVLong();
+        } else {
+            this.completionCount = -1;
```

Comment:
How about using a `Long` object and representing missing values with `null` instead of `-1`?

Reply:
I don't see any advantage of using `Long` over long here. Can you help me understand how does that add any value here.

Reply:
Mainly to better represent the optional-ness of this value.
See a similar comment here: https://github.com/opensearch-project/OpenSearch/pull/7445#discussion_r1199296121